src/backend/mpi/swdsm.cpp

```diff
@@ -901,16 +901,25 @@ void * argo_gmalloc(unsigned long size){
 	return ptrtmp;
 }
 
+/**
+ * @brief aligns an offset (into a memory region) to the beginning of its
+ * subsequent size block if it is not already aligned to a size block.
+ * @param offset the unaligned offset
+ * @param size the size of each block
+ * @return the aligned offset
+ */
+std::size_t align_forwards(std::size_t offset, std::size_t size){
+	return (offset == 0) ? offset : (1 + ((offset-1) / size))*size;
```

Comment:
The special case for `0` feels somewhat odd. Is this just to avoid the case where -1 / n would fall to 0 instead of -1? Maybe returning `0` instead of `offset` would be more clear?